os/pm/pm_checkstate.c

```diff
@@ -149,9 +149,9 @@ enum pm_state_e pm_checkstate(int domain)
 
 	/* Consider the possible power state lock here */
 
-	for (index = 0; index < pdom->recommended; index++) {
-		if (pdom->stay[index] != 0) {
-			pdom->recommended = index;
+	for(index = 0; index < CONFIG_PM_NDOMAINS; index++) {
+		if (g_pmglobals.stay[index] != 0) {
+			pdom->recommended = pdom->state;
```

Comment:
suppose state is in idle, then it will stay in idle according to this . But according to new design , we will either stay in NORMAL incase of lcd domain or STANDBY incase of other domain. Please handle that case